client/verify.go

```diff
@@ -178,20 +182,24 @@ func (v *verifyingClient) verify(ctx context.Context, info *chain.Info, r *Rando
 		}
 	}
 
-	b := chain.Beacon{
-		PreviousSig: ps,
-		Round:       r.Round(),
-		Signature:   r.Signature(),
-		SignatureV2: r.SignatureV2(),
-	}
-
 	ipk := info.PublicKey.Clone()
 	if r.Round() >= v.v2from {
+		b := chain.Beacon{
```

Comment:
style comment: we can put the beacon signature by default to be v1 and only update the field in this IF - saving us a few lines of repetition.